design/proposals/dbt-delete-merge-schema-branch.md

```diff
@@ -0,0 +1,30 @@
+# `lakectl dbt delete-branch-schema` & `lakectl dbt merge-branch-schema`
+
+## Description
+When using dbt alongside lakeFS, the first thing we would want to do is to create a branch to work on. That branch is created with a dbt schema associated with it (it might have the same name of the branch [by default] or a custom one) that is basically a copy of the schema that was associated with the branch which we branched out from but with different name and location.   
+After running dbt over that branch and conducting tests and validations we would want to either: 
+- delete the branch and the schema associated with it
+- merge them into master/main/production/other branch
+- both.
+
+`lakectl dbt delete-branch-schema` will allow the deletion of a branch and the schema associated with it, or just the schema.  
+`lakectl dbt merge-branch-schema` will allow the merging of a branch to a given branch  and the schema associated with it. 
+### Why do we need it?
+The users that use the dbt command are, well, dbt users. They will interact with the `lakectl` tool in the context of the dbt command (and some basic commands such as `commit` and `merge`).
+Adding the `delete-branch-schema` and `merge-branch-schema` subcommands will make it easier for them to integrate lakeFS into their framework.
+
+## `delete-branch-schema` Usage
```

Comment:
I think that the current usage may come out a bit confusing. The main purpose of the `delete-branch-schema` command should be to delete the **schema**. 
I see two use cases here:
1. I already deleted my dbt branch and I want to delete the schema created for that branch. 
2. I want to delete a dbt schema and its corresponding branch together. 

To support 1, `lakectl dbt delete-branch-schema --schema <schema name>` you suggested is sufficient. If we want to support 2, I suggest different wording that speaks from dbt's point of view and is consistent with the `--create-branch` flag of `create-branch-schema`. 
It can be something like something like 
`lakectl dbt delete-branch-schema  --schema <schema name> --delete-branch <branch name>`

TBH, I'm not sure that we even want to support the second use case. Let me know your thoughts and we can discuss more :)

Reply:
I agree that the primary usage of this command is to delete the schema and not the branch, but when I think of the users that will use this command, these are analysts that probably want to use mainly the `lakefs dbt` command and not the other ones... 
We could definitely create a `delete-schema` command and use it as a subcommand for the `lakectl dbt` or `lakectl metastore` commands, but then we might drift apart from the users that probably require the whole set of actions under the `lakectl dbt` command.
We can start by implementing the schema deletion part and stay tuned for branch deletion capability request (as happened with the create-branch-schema command) if shall ever come.
WDYT?